__main__.py

```diff
@@ -88,11 +79,31 @@ def on_message_sent(future):
             future.result()
 
             print()
-            loop.call_soon(prompt_and_send)
         except hangups.NetworkError:
             print('Failed to send message')
 
-    return prompt_and_send
+    return send_message
+
+
+def on_event_handler(loop, client, conversation):
+    def on_event(event):
+        """Handles the last chat message from the client."""
+
+        if isinstance(event, hangups.ChatMessageEvent):
+            loop.call_soon(show_event(event))
+            if not conversation.get_user(event.user_id).is_self:
+                loop.call_soon(
+                    send_message_callback(
+                        event,
+                        client,
+                        conversation))
+
+    def show_event(event):
+        def print_event():
+            print(event.text)
+        return print_event
```

Comment:
Minor: using the word "print" twice doesn't really aid in readability, since it's so self-explanatory, so a lambda would be useful here:

```python
def show_event(event):
    return lambda: print(event.text)
```

At that point, you could just inline the lambda straight into the `loop.call_soon` call, but I suppose you want this extra function for when we do something a little more complicated here.